services/src/main/java/edu/unc/lib/dl/cdr/services/rest/modify/IndexingController.java

```diff
@@ -15,104 +15,116 @@
+     *
      * @param id
-     * @param inplace
+     * @param inplace whether the reindex should be an in-place recursive reindex or not
      * @return
      */
     @RequestMapping(value = "edit/solr/reindex/{id}", method = RequestMethod.POST)
-    public void reindex(@PathVariable("id") String id,
-            @RequestParam(value = "inplace", required = false) Boolean inplace, HttpServletResponse response) {
-        PID pid = new PID(id);
-
-        if (!hasPermission(id)) {
-            response.setStatus(401);
-            return;
-        }
-
-        if (inplace == null || inplace) {
-            log.info("Reindexing " + id + ", inplace reindex mode");
-            operationsMessageSender.sendIndexingOperation(GroupsThreadStore.getUsername(), Arrays.asList(pid),
-                    IndexingActionType.RECURSIVE_REINDEX);
-        } else {
-            log.info("Reindexing " + id + ", clean reindex mode");
-            operationsMessageSender.sendIndexingOperation(GroupsThreadStore.getUsername(), Arrays.asList(pid),
-                    IndexingActionType.CLEAN_REINDEX);
-        }
+    public ResponseEntity<Object> reindex(@PathVariable("id") String id,
+            @RequestParam(value = "inplace", required = false) Boolean inplace) {
+        return indexObjectAndChildren(id, inplace);
     }
 
     /**
      * Perform a shallow reindexing of the object specified by id
-     * 
+     *
      * @param id
-     * @param response
      * @return
      */
     @RequestMapping(value = "edit/solr/update/{id}", method = RequestMethod.POST)
-    public void reindex(@PathVariable("id") String id, HttpServletResponse response) {
-        PID pid = new PID(id);
+    public ResponseEntity<Object> reindex(@PathVariable("id") String id) {
+        return indexObject(id);
+    }
 
-        if (!hasPermission(id)) {
-            response.setStatus(401);
-            return;
-        }
+        private ResponseEntity<Object> indexObject(String id) {
+            Map<String, Object> result = new HashMap<>();
+            result.put("action", "reindex");
+            result.put("pid", id);
 
-        log.info("Updating " + id);
-        operationsMessageSender.sendIndexingOperation(GroupsThreadStore.getUsername(),
-                Arrays.asList(pid), IndexingActionType.ADD);
-    }
+            PID objectPid = PIDs.get(id);
+
+            log.info("Updating object " + id);
+            try {
+                indexingService.reindexObject(AgentPrincipals.createFromThread(), objectPid);
+            } catch (Exception e) {
+                result.put("error", e.getMessage());
+                if (e instanceof AuthorizationException || e instanceof AccessRestrictionException) {
+                    return new ResponseEntity<>(result, HttpStatus.FORBIDDEN);
+                } else {
+                    log.error("Failed to reindex repository due to {}",  e);
```

Comment:
Could this be switched to something like `log.error("Failed to reindex object {}", id, e);`?  Also, the exception doesn't need to be formatted into the message.

Reply:
done